### PR TITLE
client: fix compat version on MStatfs

### PR DESCRIPTION
**Pull request by jcsp**
Fixes: http://tracker.ceph.com/issues/21078
Signed-off-by: John Spray <john.spray@redhat.com>

### CODE REVIEW

File: src/messages/MStatfs.h [hunk 1]
```diff
@@ -22,13 +22,13 @@
...
   static const int HEAD_VERSION = 2;
-  static const int COMPAT_VERSION = 0;
+  static const int COMPAT_VERSION = 1;
 
 public:
   uuid_d fsid;
   boost::optional<int64_t> data_pool;
 
-  MStatfs() : PaxosServiceMessage(CEPH_MSG_STATFS, 0, HEAD_VERSION) {}
+  MStatfs() : PaxosServiceMessage(CEPH_MSG_STATFS, HEAD_VERSION, COMPAT_VERSION) {}
```
Comment (batrick) [inline src/messages/MStatfs.h]: I'm not really clear on the difference between `enc_version` and `v` in:

`PaxosServiceMessage(int type, version_t v, int enc_version=1, int compat_enc_version=0)`

but based on other messages it looks like you want:

`PaxosServiceMessage(CEPH_MSG_STATFS, 0, HEAD_VERSION, COMPAT_VERSION)`

?

Comment (jcsp) [inline src/messages/MStatfs.h]: You're right, I was cribbing from another message definition that was making the same mistake!

**Comment by jcsp** [comment]
(updated)

**Comment by jcsp** [comment]
Pushed this branch to ci so that I can try installing the mon packages and re-test

**Comment by batrick** [comment]
@jcsp okay I'll standby for your test results.

**Comment by fullerdj** [comment]
Dangit, sorry about this...

**Comment by jcsp** [comment]
Tested built packages (sha1 744160d) on my mons and df works again :+1:

**Pull request merged**